x-pack/agent/pkg/agent/application/fleet_acker.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/errors"
+	"github.com/elastic/beats/x-pack/agent/pkg/core/logger"
+	"github.com/elastic/beats/x-pack/agent/pkg/fleetapi"
+	"github.com/elastic/beats/x-pack/agent/pkg/scheduler"
+)
+
+const (
+	// EventTypeAction is an event type describing action related events.
+	EventTypeAction = "ACTION"
+	// EventSubtypeACK is an ACTION event subtype specifying ACK event.
+	EventSubtypeACK = "ACKNOWLEDGED"
+)
+
+type actionAcker struct {
+	log        *logger.Logger
+	dispatcher dispatcher
+	client     clienter
+	scheduler  scheduler.Scheduler
+	agentInfo  agentInfo
+	reporter   fleetReporter
+	done       chan struct{}
+}
+
+func newActionAcker(
+	log *logger.Logger,
+	agentInfo agentInfo,
+	client clienter,
+) (*actionAcker, error) {
+	return &actionAcker{
+		log:       log,
+		client:    client,
+		agentInfo: agentInfo, //TODO(ph): this need to be a struct.
+	}, nil
+}
+
+func (f *actionAcker) Ack(actionID string) error {
+	// get events
+	event := newActionEvent(actionID)
+
+	// checkin
+	cmd := fleetapi.NewCheckinCmd(f.agentInfo, f.client)
+	req := &fleetapi.CheckinRequest{
+		Events: []fleetapi.SerializableEvent{event},
+	}
+
+	_, err := cmd.Execute(req)
+	if err != nil {
+		return errors.New(err, fmt.Sprintf("acknowledge action '%s' failed", actionID), errors.TypeNetwork)
+	}
+
+	return nil
+}
+
+func newActionEvent(actionID string) *actionEvent {
+	return &actionEvent{
+		Typ:      EventTypeAction,
+		Subtype:  EventSubtypeACK,
+		ActionID: actionID,
+		Msg:      fmt.Sprintf("Acknowledged action %s", actionID),
+	}
+}
+
+type actionEvent struct {
+	Typ      string `json:"type"`
+	Subtype  string `json:"subtype"`
+	ActionID string `json:"action_id"`
+	Msg      string `json:"message"`
+}
```

Comment:
There is already logic in place to create ACK event in the fleetapi package. see https://github.com/elastic/beats/blob/aeaae856b8710d5888d6fe42c467fe1204bf1b35/x-pack/agent/pkg/fleetapi/reply.go#L57-L63